### ISSUE TITLE
Something is wrong with instagram

### ISSUE THREAD

**Issue by erikkubica** [postaddictme/instagram-php-scraper#586]
Hey,

Fetching data from IG stopped working, the method using login fails and the method without login fails too 'Account with this username does not exist'.

It worked a while back, today client noticed that IG posts are not loaded on his website. I have also tried with my own user name and the result is the same.

Any idea?

**Comment by KinaneD** [comment]
For a while now Instagram has been trying to counter scraping its content and insights.
- The deprecation of the Instagram API
- Reduced rate limit from 5000 to 200
- A lot of reports of accounts flagged as they do not exist while they do

The general behaviour is, after hitting the rate limit numerous times the IP could be blacklisted.
We face similar issues and had to introduce a proxy that continuously shuffle our IPs. If you decide to go down this path, you've got two options, data center vs residential IPs.
Data center IPs are cheaper but are prone to this same behaviour. Residential IPs are more reliable but way more expensive.

**Comment by erikkubica** [comment]
@KinaneD thank you for the detailed explanation, I like this library simple and strait forward, but looks like we will do it the right way using the official API even if it requires more effort.

**Comment by adriandmitroca** [comment]
Hey @KinaneD 

Could you recommend some reliable proxies that are working well for you?

**Comment by KinaneD** [comment]
Hi @adriandmitroca

I'd rather not refer any Proxy provider in particular.
In general we found that Datacenter IPs have a low success rate somewhere around 40-50%. While Residential IPs are reliable.

Proxy providers differ in their offering, some offer unlimited bandwidth but allocate dedicated IPs others provide "unlimited" IPs with a limited bandwidth or any combination of these.
To use one over the other eventually depends on your application and how tolerant to failures it can be.

We went for Datacenter IPs for background processes but use Residential IPs for real-time processes and user facing operations.

Hope this helps.
Cheers.
